phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/ViewInfoWritable.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.mapreduce.util;
+
+import org.apache.hadoop.io.Writable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+public interface ViewInfoWritable extends Writable {
+    public enum ViewInfoJobState {
+        RUNNING(1),
+        SUCCEEDED(2),
+        FAILED(3),
+        PREP(4),
```

Comment:
Do u want o rename PREP to INITIALIZED?
Also maybe a good idea to keep the order (enum value) to be the state progression prep(initialized) -> RUNNING -> SUCCEEDED ....